pkg/scheduler/scheduler.go

```diff
@@ -283,9 +280,9 @@ func New(client clientset.Interface,
 
 	snapshot := internalcache.NewEmptySnapshot()
 	clusterEventMap := make(map[framework.ClusterEvent]sets.Set[string])
-	metricsRecorder := metrics.NewMetricsAsyncRecorder(1000, time.Second, stopCh)
+	metricsRecorder := metrics.NewMetricsAsyncRecorder(1000, time.Second, stopEverything)
```

Comment:
```suggestion
	metricsRecorder := metrics.NewMetricsAsyncRecorder(1000, time.Second, ctx.Done())
```

Reply:
I think it's equivalent. 
```
stopEverything := ctx.Done() 
```

The `stopEverything` variable is used in other places as well(Line 326,330,339).
We don't need to call `ctx.Done()` again.